File: 0205-isomorphic-strings/0205-isomorphic-strings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function (s, t) {
    if (s.length !== t.length) return false
    let map1 = {}
    let map2 = {}
    for (let i = 0; i < s.length; i++) {
        let sChar = s[i]
        let tChar = t[i]

        if (map1[sChar] && map1[sChar] !== tChar) {
            return false;
        }
        if (map2[tChar] && map2[tChar] !== sChar) {
            return false;
        }
        map1[sChar] = tChar
        map2[tChar] = sChar

    }
    return true;
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False
        map1 = {}
        map2 = {}
        for i in range(len(s)):
            sChar = s[i]
            tChar = t[i]
            if sChar in map1 and map1[sChar] != tChar:
                return False 
            if tChar in map2 and map2[tChar] != sChar:
                return False
            map1[sChar] = tChar
            map2[tChar] = sChar
        return True



        